pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -72,12 +73,39 @@ func (ds *dockerService) RunPodSandbox(config *runtimeApi.PodSandboxConfig) (str
 	// Assume kubelet's garbage collector would remove the sandbox later, if
 	// startContainer failed.
 	err = ds.StartContainer(createResp.ID)
+	if err != nil {
+		return createResp.ID, fmt.Errorf("failed to start sandbox container for pod %q: %v", config.Metadata.GetName(), err)
+	}
+
+	// Step 4: Trigger network plugin to setup pod network
+	// skip if pod is using hostnetwork
+	if !config.GetLinux().GetNamespaceOptions().GetHostNetwork() {
+		glog.V(2).Infof("Calling network plugin %q to setup network for pod %s/%s", ds.networkPlugin.Name(), config.GetMetadata().GetNamespace(), config.GetMetadata().GetName())
+		err = ds.networkPlugin.SetUpPod(config.GetMetadata().GetNamespace(), config.GetMetadata().GetName(), createResp.ID)
+		if err != nil {
+			// Best Effort clean up sandbox
+			if cleanUpErr := ds.StopPodSandbox(createResp.ID); cleanUpErr != nil {
+				glog.Errorf("failed to clean up sandbox %q for pod %q: %v", createResp, config.Metadata.GetName(), cleanUpErr)
+			}
+			return createResp.ID, fmt.Errorf("failed to call network plugin %q to setup sandbox newtork for pod %q: %v", ds.networkPlugin.Name(), config.Metadata.GetName(), err)
+		}
+	}
 	return createResp.ID, err
 }
 
 // StopPodSandbox stops the sandbox. If there are any running containers in the
 // sandbox, they should be force terminated.
 func (ds *dockerService) StopPodSandbox(podSandboxID string) error {
+	status, err := ds.PodSandboxStatus(podSandboxID)
+	if err != nil {
+		glog.Errorf("failed to get pod sandbox status: %v", err)
+	}
+	if status != nil && !status.GetLinux().GetNamespaces().GetOptions().GetHostNetwork() {
```

Comment:
> !status.GetLinux().GetNamespaces().GetOptions().GetHostNetwork()

need checks nil pointer here too.

> so we inspect the container id and plug some pieces of information out of it. Does this work for all required
> information?

I think only hostNetwork mode is required here.